### PR TITLE
sample_decode: replace MSDKThread with std::thread

### PR DESCRIPTION
**Pull request by mgonchar**

### CODE REVIEW

File: samples/sample_decode/src/pipeline_decode.cpp [hunk 1]
```diff
@@ -1660,21 +1655,18 @@ mfxStatus CDecodingPipeline::RunDecoding()
...
-        if (!pDeliverThread || !m_pDeliverOutputSemaphore || !m_pDeliveredEvent) {
-            MSDK_SAFE_DELETE(pDeliverThread);
-            MSDK_SAFE_DELETE(m_pDeliverOutputSemaphore);
-            MSDK_SAFE_DELETE(m_pDeliveredEvent);
-            return MFX_ERR_MEMORY_ALLOC;
-        }
+
+        // Create a thread and let it work independently from the thread handle 
+        std::thread deliverThread(&CDecodingPipeline::DeliverLoop, this);
+        deliverThread.detach();
```
Comment (dvrogozh) [inline samples/sample_decode/src/pipeline_decode.cpp]: hm. why we detach the thread? what is your motivation for that? Will this make memory leaks checking more complicated? IMHO, I would prefer to see honest join in the very end.

File: samples/sample_decode/src/pipeline_decode.cpp [hunk 2]
```diff
@@ -1660,21 +1655,18 @@ mfxStatus CDecodingPipeline::RunDecoding()
     bool                bErrIncompatibleVideoParams = false;
     CTimeInterval<>     decodeTimer(m_bIsCompleteFrame);
     time_t start_time = time(0);
-    MSDKThread * pDeliverThread = NULL;
 #if (MFX_VERSION >= 1025)
     mfxExtDecodeErrorReport *pDecodeErrorReport = NULL;
 #endif
 
-    if (m_eWorkMode == MODE_RENDERING) {
+    if (m_eWorkMode == MODE_RENDERING)
```
Comment (dvrogozh) [inline samples/sample_decode/src/pipeline_decode.cpp]: please, don't change what you actually did not modify.

File: samples/sample_decode/src/pipeline_decode.cpp [hunk 3]
```diff
@@ -1963,16 +1955,14 @@ mfxStatus CDecodingPipeline::RunDecoding()
             CTimer::ConvertToSeconds(*std::min_element(m_vLatency.begin(), m_vLatency.end()))*1000);
     }
 
-    if (m_eWorkMode == MODE_RENDERING) {
+    if (m_eWorkMode == MODE_RENDERING)
```
Comment (dvrogozh) [inline samples/sample_decode/src/pipeline_decode.cpp]: please, don't change what you actually did not modify.

File: samples/sample_decode/src/pipeline_decode.cpp (hunk 1 above)
Comment (mgonchar) [reply samples/sample_decode/src/pipeline_decode.cpp]: Well, ```join``` looks more similar to original code, so I redesigned it with ```join```

File: samples/sample_decode/src/pipeline_decode.cpp (hunk 2 above)
Comment (mgonchar) [reply samples/sample_decode/src/pipeline_decode.cpp]: fixed

File: samples/sample_decode/src/pipeline_decode.cpp (hunk 3 above)
Comment (mgonchar) [reply samples/sample_decode/src/pipeline_decode.cpp]: fixed

File: samples/sample_decode/src/pipeline_decode.cpp [hunk 4]
```diff
@@ -1559,17 +1558,7 @@ mfxStatus CDecodingPipeline::DeliverLoop(void)
...
-
-unsigned int MFX_STDCALL CDecodingPipeline::DeliverThreadFunc(void* ctx)
-{
-    CDecodingPipeline* pipeline = (CDecodingPipeline*)ctx;
-
-    mfxStatus sts;
-    sts = pipeline->DeliverLoop();
-
-    return 0;
+    return;
```
Comment (dvrogozh) [inline samples/sample_decode/src/pipeline_decode.cpp]: minor. 'return' is useless here - drop.

Comment (mgonchar) [reply samples/sample_decode/src/pipeline_decode.cpp]: will do in next changes, I plan to get rid of of semaphores here
